source/includes/page-templates/update/arrays.rst

```diff
@@ -0,0 +1,370 @@
+       **Data Type:** ``Expression<Func<TDocument, IEnumerable<TItem>>>``
+
+   * - ``value``
+     - The value to add to the end of the array field.
+
+       **Data Type:** ``TItem`` 
+
+The following code example calls the ``AddToSet()`` method to re-add the first
+``GradeEntry`` object to the ``Grades`` array in |matching-document-or-documents|. Because
+the value already exists in the array, the update operation does nothing.
+
+|addtoset-code-example-tabs|
+
+Add Multiple Values
+-------------------
+
+To add multiple values to an array, call the ``Builders.Update.PushEach()`` method.
+This method accepts the following parameters:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+
+   * - ``field``
+     - An expression that specifies the array field to add one or more values to.
+
+       **Data Type:** ``Expression<Func<TDocument, IEnumerable<TItem>>>``
+
+   * - ``values``
+     - The values to add to the array field.
+
+       **Data Type:** ``IEnumerable<TItem>``
+
+   * - ``slice``
+     - The number of elements to keep in the array, counted from the start of the array
+       after updates. If the value is negative,
+       the method keeps the specified number of elements from the end of the array.
+
+       **Data Type:** ``int?``
+
+   * - ``position``
+     - The position in the array at which to add the values. By default, the method
+       adds the values to the end of the array.
+
+       **Data Type:** ``int?``
+
+   * - ``sort``
+     - A ``SortDefinition`` object that specifies how the driver sorts the array elements
+       after adding the new values.
+
+       **Data Type:** `SortDefinition<TItem> <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.SortDefinition-1.html>`__
+
+The following code example uses the ``PushEach()`` method to add two new ``GradeEntry``
+objects to the start of the ``Grades`` array in |matching-document-or-documents|.
+It then sorts the array elements in descending order by the values of their ``Score`` fields.
+
+|pusheach-code-example-tabs|   
+
+To add multiple values to an array, *but only if they don't already exist in the array*,
+call the ``Builders.Update.AddToSetEach()`` method.
+This method accepts the following parameters:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+
+   * - ``field``
+     - An expression that specifies the array field to add one or more values to.
+
+       **Data Type:** ``Expression<Func<TDocument, IEnumerable<TItem>>>``
+
+   * - ``values``
+     - The values to add to the array field.
+
+       **Data Type:** ``IEnumerable<TItem>``
+
+The following code example calls the ``AddToSetEach()`` method to re-add the first
+and second ``GradeEntry`` objects to the ``Grades`` array in |matching-document-or-documents|.
+Because these values already exist in the array, the update operation does nothing.
+
+|addtoseteach-code-example-tabs| 
+
+Remove Values
+-------------
```

Comment:
Not a problem to do this, but I used subheaders in the later section because they all use the same method. LMK what you think about the subheaders in this section